plugins/signals-backend/src/service/router.ts

```diff
@@ -80,23 +88,32 @@ export async function createRouter(
 
       // Authentication token is passed in Sec-WebSocket-Protocol header as there
       // is no other way to pass the token with plain websockets
-      const token = req.headers['sec-websocket-protocol'];
-      if (token) {
-        userIdentity = await identity.getIdentity({
-          request: {
-            headers: { authorization: token },
-          },
-        } as IdentityApiGetIdentityRequest);
+      try {
+        const token = req.headers['sec-websocket-protocol'];
+        if (token) {
+          userIdentity = await identity.getIdentity({
+            request: {
+              headers: { authorization: `Bearer ${token}` },
+            },
+          } as IdentityApiGetIdentityRequest);
+        }
+      } catch (e) {
+        logger.error('Failed to authenticate WebSocket connection', e);
+        return;
```

Comment:
What's the difference in behavior for the caller here? I presume this ends up being a 404, which might not be quite what we want?

Reply:
Hm, maybe. Should we just open the connection with unknown user info in that case?

Reply:
Something like this could be done but not sure if it's ok:
```ts
socket.write('HTTP/1.1 401 Web Socket Protocol Handshake\r\n' +
                 'Upgrade: WebSocket\r\n' +
                 'Connection: Upgrade\r\n' +
                 '\r\n');
socket.close();
socket.destroy();
return;
```